File: ColorSudoku/src/main/java/sk/tuke/gamestudio/service/RatingService.java
package sk.tuke.gamestudio.service;

import sk.tuke.gamestudio.entity.Rating;

import java.util.List;

public interface RatingService {
    void setRating(Rating rating) throws RatingException;
    int getAverageRating(String game) throws RatingException;
    int getRating(String game, String player) throws RatingException;
    List<Rating> getRatingPlayer(String game) throws RatingException;
    Rating getRatingClass(String game, String player) throws RatingException;
    void reset() throws RatingException;
}

File: frontend/src/_api/rating.service.js
import {_exios} from "./index";
import {formatDate} from "./utils";

export const addRating = function (game, player, rating){
     return _exios.post('/rating', {
        game, player, rating, ratedOn : formatDate(new Date)
    })
};

export const getAverageRating = function(game){
    return _exios.get('/rating/average/' + game);
};

export const getRating = function(game, player){
    return _exios.get('/rating/onerating/' + game +'/' + player);
};

export const getRatingPlayer = function (game){
    return _exios.get('/rating/ratinglist/' + game);
};

export const getRatingClass = function(game, player){
    return _exios.get("/rating/getratingclass/" + game + '/' + player);
};